hh-dep-monitoring-service/src/main/java/ru/hh/school/depmonitoring/dao/AbstractDao.java

```diff
@@ -0,0 +1,63 @@
+package ru.hh.school.depmonitoring.dao;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+public abstract class AbstractDao<T> implements Dao<T> {
+
+    private Class<T> clazz;
+    protected SessionFactory sessionFactory;
+
+    public AbstractDao(SessionFactory sessionFactory, Class<T> clazz) {
+        this.sessionFactory = sessionFactory;
+        this.clazz = clazz;
+    }
+
+    protected Session getSession() {
+        return sessionFactory.getCurrentSession();
+    }
+
+    public final void setClazz(Class<T> clazzToSet) {
+        this.clazz = clazzToSet;
+    }
+
+    @Transactional
+    public Optional<T> findOne(int id) {
+        return (Optional<T>) getCurrentSession().get(clazz, id);
+    }
+
+    @Transactional
+    public List<T> findAll() {
+        return getCurrentSession().createQuery("from " + clazz.getName()).list();
+    }
+
+    @Transactional
+    public void create(Optional<T> entity) {
```

Comment:
не входной параметр не должен быть Optional
```suggestion
    public void create(@Nonnull T entity) {
```

Reply:
ок